app/models/manager_refresh/graph/topological_sort.rb

```diff
@@ -46,14 +43,14 @@ def make_topological_sort(original_nodes, edges)
           i         += 1
           max_depth -= 1
           if max_depth <= 0
-            raise "Max depth reached while doing topological sort of nodes #{original_nodes} and edges #{edges}, your "\
-                  "graph probably contains a cycle"
+            raise "Max depth reached while doing topological sort, your graph probably contains a cycle:\n"\
+                  "#{graph.to_graphviz}"
```

Comment:
Can you log the graphviz output to debug instead of dumping it in the exception message?

Reply:
Done except log.error, as we do for exception-adjacent logging like backtraces.
But mostly because IMHO not having the info because debug was hidden would be a pity...

Reply:
You're right logged to error is perfect, mostly I wanted it out of the exception since we show refresh exceptions on the UI.